src/app/components/change-rep-widget/change-rep-widget.component.ts

```diff
@@ -37,7 +39,19 @@ export class ChangeRepWidgetComponent implements OnInit {
       this.updateDisplayedRepresentatives();
     });
 
-    await this.repService.detectChangeableReps();
+    // Detect if a wallet is reset
+    this.walletService.wallet.newWallet$.subscribe(async bool => {
+      if (bool) {
+        console.log('Reloading representatives..');
+        this.selectedAccount = null;
+        this.representatives = [];
+        this.changeableRepresentatives = [];
+        this.showRepChangeRequired = false;
+        this.updateDisplayedRepresentatives();
+        this.representatives = await this.repService.getRepresentativesOverview(); // calls walletReps$.next
```

Comment:
This could get moved to a new function like `resetRepresentatives` or so.

Reply:
done